FAQ.md

```diff
@@ -59,3 +59,63 @@ error:
 ```
 then specify the package you want in the import, e.g:
 `import "cryptonite" Crypto.Hash`
+
+
+### What does `ob run` actually do?
+
+
+#### Short version:
+
+`ob run` starts a [`ndmitchell/ghcid`](https://github.com/ndmitchell/ghcid)
+process, within a carefully crafted `nix-shell`, which tries to build your
+project and,
+
+* either displays compilation errors,
+* or starts the Obelisk server, which serves:
+     * your backend's route handlers,
+     * the static assets,
+     * the [JSaddle](https://github.com/ghcjs/jsaddle) frontend code (while opening its websocket).
+
+#### Longer version:
+
+Assuming we are in a project created with `ob init`, `ob run` calls (see
+`./lib/command/src/Obelisk/Command.hs`):
+
+    nix-shell -A shells.ghc --run 'ob --no-handoff internal run-static-io <real-run-function>'
+
+where
+
+* `shells.ghc` is defined in `./default.nix` by importing `./.obelisk/impl/default.nix` which
+   is `./default.nix` in the present (Obelisk) repository,
+* `run-static-io` is a logging-enabled command wrapper
+   (cf. `runObelisk` in `./lib/command/src/Obelisk/App.hs`).
+
+In this case, it runs the function `Obelisk.Command.Command.run` (defined in
+`./lib/command/src/Obelisk/Command/Run.hs`).
+
+* It creates a GHCid config from a Nix expression:
+    * which loads 3 packages: "backend", "common", "frontend", and
+    * obtains a free port number.
+* Then runs `ghcid`
+    * with a `--test` command that reruns `Obelisk.Run.run` at each restart.
+
+It is defined in `./obelisk/lib/run/src/Obelisk/Run.hs`:
+
+* It creates a thread (`forkIO`) which starts the main backend.
+  This runs `runSnapWithCommandLineArgs` and
+    * passes routes to the backend
+      (result of the `_backend_run` field of the `Backend x y` “user” record)
```

Comment:
Are the type parameters just called `x` and `y` in the definition of the record? At least here they should be renamed to be semantically clear. Also I'm not sure "user" is necessary.